explain-subqueries.md

```diff
@@ -0,0 +1,146 @@
+
+The query is said to be a _correlated subquery_ because the column `t1.int_col` is pushed directly into the subquery, but actually from the `EXPLAIN` output we can see that it is not correlated at all. This is because TiDB has rewritten `t1_id != t1.int_col` to `t1.id != t1.int_col` and moved it out of the subquery. This is a lot more efficient to process, because TiDB can perform this check in `└─Selection_21` as it is reading from the table `t1`.
+
+## Anti semi join (`NOT IN` subquery)
+
+In the following example, the query semantically returns all rows from the table `t3` _unless_ `t3.t1_id` is in the subquery:
+
+```sql
+EXPLAIN SELECT * FROM t3 WHERE t1_id NOT IN (SELECT id FROM t1 WHERE int_col < 100);
+```
+
+```sql
++-----------------------------+---------+-----------+---------------+-------------------------------------------------------------------------------------+
+| id                          | estRows | task      | access object | operator info                                                                       |
++-----------------------------+---------+-----------+---------------+-------------------------------------------------------------------------------------+
+| IndexMergeJoin_20           | 1598.40 | root      |               | anti semi join, inner:TableReader_15, outer key:test.t3.t1_id, inner key:test.t1.id |
+| ├─TableReader_28(Build)     | 1998.00 | root      |               | data:TableFullScan_27                                                               |
+| │ └─TableFullScan_27        | 1998.00 | cop[tikv] | table:t3      | keep order:false                                                                    |
+| └─TableReader_15(Probe)     | 1.00    | root      |               | data:Selection_14                                                                   |
+|   └─Selection_14            | 1.00    | cop[tikv] |               | lt(test.t1.int_col, 100)                                                            |
+|     └─TableRangeScan_13     | 1.00    | cop[tikv] | table:t1      | range: decided by [test.t3.t1_id], keep order:true                                  |
++-----------------------------+---------+-----------+---------------+-------------------------------------------------------------------------------------+
+6 rows in set (0.00 sec)
+```
+
+This query starts by reading the table `t3`, and then probes the table `t1` based on the `PRIMARY KEY`. The join type is an _anti semi join_; anti because the test is for the non-existence of the value (`NOT IN`) and semi-join because only the first row needs to match before the join is rejected.
```

Comment:
Is there any difference between `PRIMARY KEY` and primary key?
```suggestion
This query starts by reading the table `t3` and then probes the table `t1` based on the `PRIMARY KEY`. The join type is an _anti semi join_; anti because this example is for the non-existence of the value (`NOT IN`) and semi-join because only the first row needs to match before the join is rejected.
```